apollo-router/src/spec/query.rs

```diff
@@ -497,11 +596,31 @@ impl Query {
                                     *output = input.clone();
                                     Ok(())
                                 } else {
+                                    parameters.validation_errors.push(
+                                        Error::builder()
+                                            .message(format!(
+                                                "Expected a valid enum value for type {}",
+                                                enum_type.name
+                                            ))
+                                            .path(Path::from_response_slice(path))
+                                            .extension_code(RESPONSE_VALIDATION)
+                                            .build(),
+                                    );
                                     *output = Value::Null;
                                     Ok(())
                                 }
                             }
                             None => {
+                                parameters.validation_errors.push(
```

Comment:
I think there's a missing `if !input.is_null()` or something here. If I understand correctly, the `executable::Type::Named(type_name)` matcher arm that this is under actually means "nullable named type" (vs `executable::Type::NonNullNamed`), which means it is valid for `input` to be null.  This change is leading to the addition of an incorrect `Expected a valid enum value for type {}` error to responses any time that a subgraph returns `null` for a nullable enum field.